webapp/src/components/form/FormCreator.js

```diff
@@ -0,0 +1,310 @@
+ 
+
+
+const FormCreator = ({
+  languages, event: evt, t, sections,
+  setSections, nominate, setNominate,
+  language, setLanguage, dragId, setDragId,
+  applyTransition, setApplytransition,
+  parentDropable, setParentDropable,
+  homeRedirect, initialState, errorResponse,
+  disableSaveBtn, setDisableSaveBtn,
+  events, setEvent, eventService, addSection,
+  handleSave, isSaved, isReview, addQuestion,
+  addAnswerFromAppForm, appSections, stage,
+  setCurrentStage, currentStage, leaveStage,
+  setLeaveStage, showingModal, setShowingModal,
+  isNewStage, setIsNewStage
+}) => {
+  const lang = languages;
+
+  const saved = _.isEqual(initialState, sections);
+  let maxSurrogateId = 1;
+  sections.forEach(s => {
+    if (s.backendId > maxSurrogateId) {
+      maxSurrogateId = s.backendId;
+    }
+    s.questions.forEach(q => {
+      if (q.backendId > maxSurrogateId) {
+        maxSurrogateId = q.backendId
+      }
+      if (q.surrogate_id > maxSurrogateId) {
+        maxSurrogateId = q.surrogate_id
+      }
+    })
+  })
+
+  useEffect(() => {
+    const eventId = evt.id;
+    eventService.getEvent(eventId).then( res => {
+      setEvent({
+        loading: false,
+        event: res.event,
+        error: res.error
+      })
+    });
+  }, []);
+
+  const handleCheckChanged = (e) => {
+    const val = e.target.checked;
+    setTimeout(() => {
+      setNominate(val);
+    }, 1);
+  }
+
+  const handleLanguageChange = (e) => {
+    setLanguage(e);
+  }
+
+  const handleSection = (input) => {
+    setSections(input);
+  }
+
+  const handleDrag = (e) => {
+    if(parentDropable) {
+      drag(e, setDragId);
+    }
+  }
+
+  const handleDragOver = (e) => {
+    e.preventDefault();
+  };
+
+  const handleDrop = (e) => {
+    if (parentDropable) {
+      drop({
+        event: e,
+        elements: sections,
+        dragId,
+        setState: setSections,
+        setAnimation: setApplytransition
+      });
+    }
+  }
+
+  const options = () => {
+    return lang.map(l => option({
+      value: l.code,
+      label: l.description,
+      t
+    }));
+  }
+
+  let isSaveDisabled = false;
+  sections.forEach(s => {
+    for(let key of Object.keys(s.name)) {
+      if (!s.name[key]) {
+        isSaveDisabled = true;
+      }
+    }
+    s.questions.forEach(q => {
+      for(let key of Object.keys(q.headline)) {
+        if (!q.headline[key]) {
+          isSaveDisabled = true;
+        }
+      }
+      if (!q.type) {
+        isSaveDisabled = true;
+      }
+    })
+  })
+
+  
+
+  const {loading, event: evnt, error } = events;
+
+  if (loading) {
+    return <Loading />
+  }
+  if (error) {
+    return (
+      <div className='alert alert-danger alert-container'>
+        {error}
+      </div>
+    )
+  }
+  return (
+    <>
+      <Prompt
+        when={!isSaved && !saved}
+        message="Some Changes have not been saved. Are you sure you want to leave without saving them?"
+        />
+      {homeRedirect && <Redirect to="/" />}
```

Comment:
Can we redirect to the event home page instead of the main home page? When using the application form builder I found it a bit confusing to go back to the main home page, when I started on the event home page.